service/service.go

```diff
@@ -0,0 +1,310 @@
+
+func NewGnomobileService(opts ...GnomobileOption) (GnomobileService, error) {
+	svc := &gnomobileService{}
+	if err := svc.applyOptions(opts...); err != nil {
+		return nil, err
+	}
+
+	svc.client = NewClient(Opts{
+		Remote:  svc.remote,
+		ChainID: svc.chainID,
+	})
+
+	if err := svc.client.InitKeyBaseFromDir(svc.rootDir); err != nil {
+		return nil, err
+	}
+
+	// start gRPC server
+	svc.runServer()
+
+	return svc, nil
+}
+
+func (s *gnomobileService) applyOptions(opts ...GnomobileOption) error {
+	withDefaultOpts := make([]GnomobileOption, len(opts))
+	copy(withDefaultOpts, opts)
+	withDefaultOpts = append(withDefaultOpts, WithFallbackDefaults)
+	for _, opt := range withDefaultOpts {
+		if err := opt(s); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *gnomobileService) runServer() error {
+	// delete socket if it already exists
+	if _, err := os.Stat(s.socketPath); !os.IsNotExist(err) {
+		if err := os.RemoveAll(s.socketPath); err != nil {
+			return err
+		}
+	}
+
+	listener, err := net.Listen("unix", s.socketPath)
+	if err != nil {
+		return err
+	}
+
+	server := grpc.NewServer()
+
+	gnomobiletypes.RegisterGnomobileServiceServer(server, s)
+	go func() {
+		// we dont need to log the error
+		_ = server.Serve(listener)
+	}()
+
+	s.listener = listener
+	s.server = server
+
+	return nil
+}
+
+func (s *gnomobileService) GetSocketPath() string {
+	return s.socketPath
+}
+
+func (s *gnomobileService) Close() error {
+	if s.server != nil {
+		s.server.Stop()
+	}
+
+	return nil
+}
+
+type GnomobileOption func(*gnomobileService) error
+
+// FallBackOption is a structure that permit to fallback to a default option if the option is not set.
+type FallBackOption struct {
+	fallback func(s *gnomobileService) bool
+	opt      GnomobileOption
+}
+
+// --- Logger options ---
+
+// WithLogger set the given logger.
+var WithLogger = func(l *zap.Logger) GnomobileOption {
+	return func(s *gnomobileService) error {
+		s.logger = l
+		return nil
+	}
+}
+
+// WithDefaultLogger init a noop logger.
+var WithDefaultLogger GnomobileOption = func(s *gnomobileService) error {
+	logger, err := zap.NewDevelopment()
+	if err != nil {
+		return err
+	}
+
+	s.logger = logger
+
+	return nil
+}
+
+var fallbackLogger = FallBackOption{
+	fallback: func(s *gnomobileService) bool { return s.logger == nil },
+	opt:      WithDefaultLogger,
+}
+
+// WithFallbackLogger set the logger if no logger is set.
+var WithFallbackLogger GnomobileOption = func(s *gnomobileService) error {
+	if fallbackLogger.fallback(s) {
+		return fallbackLogger.opt(s)
+	}
+	return nil
+}
+
+// --- Remote options ---
+
+// WithRemote set the given remote node address.
+var WithRemote = func(remote string) GnomobileOption {
+	return func(s *gnomobileService) error {
+		s.remote = remote
+		return nil
+	}
+}
+
+// WithDefaultRemote init a default remote node address.
+var WithDefaultRemote GnomobileOption = func(s *gnomobileService) error {
+	s.remote = "testnet.gno.berty.io:26657"
```

Comment:
What is the difference between this default remote and the one in the Client struct?
https://github.com/gnolang/gnomobile/blob/a07f32725fef3ec12c256445659d00afb81ade2e/service/client.go#L36